### PR TITLE
[rstmgr] Update reset info to include software reset

### PR DESCRIPTION
**Pull request by tjaychen**
- Addresses #10383

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

File: hw/ip/rstmgr/doc/_index.md [hunk 1]
```diff
@@ -284,31 +284,33 @@ Reset Cause             | Description
...
 
 
 The reset info register is write 1 clear.
 It is software responsibility to clear old reset reasons; the reset manager simply records based on the rules below.
 
 Excluding power on reset, which is always recorded when the device POR circuitry is triggered, the other resets are recorded when authorized by the reset manager.
-Reset manager authorization in turn is based on reset category as indicated by the power manager.
-The power manager observes 3 categories of reset states that are mutually exclusive.
+Reset manager authorization in turn is based on reset categories as indicated by the power manager.
+The power manager observes 4 categories of reset states that are mutually exclusive.
```
Comment (matutem) [inline hw/ip/rstmgr/doc/_index.md:296]:
```suggestion
The power manager observes four categories of reset states that are mutually exclusive.
```
To be consistent with the text following.

File: hw/ip/rstmgr/doc/_index.md [hunk 2]
```diff
@@ -284,31 +284,33 @@ Reset Cause             | Description
...
 *   No resets are triggered through the power manager.
```
Comment (matutem) [inline hw/ip/rstmgr/doc/_index.md:297]: I don't see why "No " here. I assume this refers to resets that are triggered by pwrmgr, like power glitch or escalation.

File: hw/ip/rstmgr/doc/_index.md [hunk 3]
```diff
@@ -284,31 +284,33 @@ Reset Cause             | Description
...
 *   Resets triggered by low power entry.
+*   Resets triggered by software.
 *   Resets triggered by a peripheral request.
 
-Whenever the power manager begins one of the latter two sequences, it sends a hint to the reset manager so that the reset manager can decide which reason to record when the processor reset is observed.
+Whenever the power manager begins one of the latter three sequences, it sends a hint to the reset manager so that the reset manager can decide which reason to record when the processor reset is observed.
 Whenever the power manager is NOT in one of the two latter states, non-debug-module resets are allowed and directly handled and recorded by the reset manager.
```
Comment (matutem) [inline hw/ip/rstmgr/doc/_index.md:303]:
```suggestion
Whenever the power manager is NOT in one of the three latter states, non-debug-module resets are allowed and directly handled and recorded by the reset manager.
```

File: hw/ip/rstmgr/doc/_index.md [hunk 4]
```diff
@@ -284,31 +284,33 @@ Reset Cause             | Description
...
 
 Since a reset could be motivated by multiple reasons (a security escalation during low power transition for example), the reset information registers constantly record all reset causes in which it is allowed.
 The only case where this is not done is `POR`, where active recording is silenced until the first processor reset release.
 
-Despite 3 reset causes all labeled as warm boot, their effects on the system are not identical.
+Despite 4 reset causes all labeled as warm boot, their effects on the system are not identical.
```
Comment (matutem) [inline hw/ip/rstmgr/doc/_index.md:308]:
```suggestion
Despite four reset causes are labeled as warm boot, their effects on the system are not identical.
```

**Review by matutem — APPROVED** [review]
With some simple suggestions.

File: hw/ip/rstmgr/doc/_index.md (hunk 2 above)
Comment (tjaychen) [reply hw/ip/rstmgr/doc/_index.md:297]: i hvae no idea what i was trying to say there. Let me reword.

Comment (tjaychen) [reply hw/ip/rstmgr/doc/_index.md:297]: o i remember now, i was trying to say one of the cases is that no resets are triggered through the pwrmgr (ndm_rst)

File: hw/ip/rstmgr/doc/_index.md [hunk 5]
```diff
@@ -284,31 +284,32 @@ Reset Cause             | Description
...
 
 Excluding power on reset, which is always recorded when the device POR circuitry is triggered, the other resets are recorded when authorized by the reset manager.
-Reset manager authorization in turn is based on reset category as indicated by the power manager.
-The power manager observes 3 categories of reset states that are mutually exclusive.
-*   No resets are triggered through the power manager.
-*   Resets triggered by low power entry.
-*   Resets triggered by a peripheral request.
+Reset manager authorization is based on reset categories as indicated by the power manager.
+The power manager observes three categories of reset states that are mutually exclusive.
+*   No reset triggered through the power manager.
```
Comment (matutem) [inline hw/ip/rstmgr/doc/_index.md:297]: This is rather awkward: you are saying that power manager observes these, but one of them is not even seen by pwrmgr. Maybe you mean "The reset manager observes...".

Also, the first bullet item can be "Reset not triggered through the power manager."

File: hw/ip/rstmgr/doc/_index.md [hunk 6]
```diff
@@ -284,31 +284,32 @@ Reset Cause             | Description
...
+*   Reset triggered by low power entry.
+*   Reset triggered by software or a peripheral request.
 
 Whenever the power manager begins one of the latter two sequences, it sends a hint to the reset manager so that the reset manager can decide which reason to record when the processor reset is observed.
-Whenever the power manager is NOT in one of the two latter states, non-debug-module resets are allowed and directly handled and recorded by the reset manager.
+Whenever the power manager is NOT in one of the two latter states, non-debug-module resets are allowed and directly recorded by the reset manager.
```
Comment (matutem) [inline hw/ip/rstmgr/doc/_index.md:302]: Here you are explicitly saying the resets not triggered through power manager are non-debug module resets. Perhaps you can make that explicit in that bullet item instead.

**Review by matutem — APPROVED** [review]
With a couple suggestions. LGTM.

File: hw/ip/rstmgr/doc/_index.md (hunk 5 above)
Comment (tjaychen) [reply hw/ip/rstmgr/doc/_index.md:297]: sure...

File: hw/ip/rstmgr/doc/_index.md (hunk 6 above)
Comment (tjaychen) [reply hw/ip/rstmgr/doc/_index.md:302]: sure...

Comment (tjaychen) [reply hw/ip/rstmgr/doc/_index.md:302]: no that is not what i am saying.

**Comment by tjaychen** [comment]
hey @matutem please have a look at the latest and if it's better.

**Pull request merged**